app/controllers/api/v1/posts_controller.rb

```diff
@@ -7,30 +7,39 @@ class PostsController < Api::V1::BaseController
       before_action :find_post, only: %i[show update]
 
       def index
+        return render json: { message: 'User does not exist' }, status: :bad_request unless @user.present?
+
         @posts = @user.posts.order(published_at: :desc)
       end
 
-      def show; end
+      def show
+        render json: { message: 'Post does not exist' }, status: :bad_request unless @post.present?
+      end
 
       def create
+        return render json: { message: 'Unauthorized' }, status: :unauthorized unless @user.present?
+
         @post = @user.posts.build(post_params)
-        @post.save if validate_post(@user, @post)
+        @post.save if validate_post(@user, @post.title, @post.body)
       end
 
       def update
-        return unless validate_not_empty_attriutes(post_params[:title], post_params[:body])
+        return render json: { message: 'Unauthorized' }, status: :unauthorized unless @post.present?
+
+        new_post = params[:post]
+        return unless validate_post(@post.user, new_post[:title], new_post[:body])
 
         @post.update(post_params)
         render 'api/v1/posts/create', formats: [:json], locals: { post: @post }
       end
 
       # :reek:ControlParameter
-      def validate_post(user, post)
+      def validate_post(user, title, body)
         if user != current_user
           render json: { message: 'Unauthorized' }, status: :unauthorized
           return false
         end
```

Comment:
And then you can remove this validation, and just call `validate_not_empty_attriutes` directly